cohortextractor/concepts/types.py

```diff
@@ -1,15 +1,31 @@
 class BaseType:
-    pass
+    """
+    Base class for Column types defined in a Contract
+    """
+
+    # Subclasses must specify the backend Column types that they allow, as
+    # tuples of one or more keys from cohortextractor.sqlalchemy_types.TYPES_BY_NAME
+    allowed_backend_types: tuple = NotImplemented
```

Comment:
In the commit message you note that this is "a tuple of one or more keys from `sqlalchemy_types.TYPES_BY_NAME`", but we don't assert that anywhere. Maybe we could check it in `__init_subclass_()`? I fear that otherwise spelling mistakes in this field will cause hard-to-debug errors.

Or, maybe even better, we could make the keys of `TYPES_BY_NAME` an enumeration and use that here (and in the backend definitions).

Reply:
Yes, I wondered about both of these things.  I had an Enum initially and then simplified it for the first implementation, but I'll look at it again